File: frontend/src/api/oidcConnect.ts
import api from './client'

import type { OIDCPreset } from './settings'

export interface ConnectStartResponse {
  redirect_url: string
}

export interface OIDCLinkItem {
  provider_id: string
  provider_name: string
  preset: OIDCPreset
  sub_hint: string
}

export interface OIDCLinkResponse {
  link: OIDCLinkItem | null
}

export function getOIDCLink() {
  return api.get<OIDCLinkResponse>('/account/oidc/links')
}

export function startConnect(providerId: string) {
  return api.post<ConnectStartResponse>(
    `/account/oidc/connect/start/${providerId}`,
    {},
  )
}

export function disconnectOIDC() {
  return api.delete('/account/oidc/links')
}

File: backend/app/schemas/oidc_connect.py
"""Schemas for the authed OIDC connect flow (Phase 10)."""
from __future__ import annotations

from ..models.oidc_provider import OIDCPreset
from .common import APIBaseModel


class ConnectStartResponse(APIBaseModel):
    redirect_url: str


class OIDCLinkItem(APIBaseModel):
    provider_id: str
    provider_name: str
    preset: OIDCPreset
    sub_hint: str  # truncated subject for display ("…1234")


class OIDCLinkResponse(APIBaseModel):
    """Returned by GET /api/account/oidc/links - at most one link per
    user, reflecting the locked single-provider-per-user design."""
    link: OIDCLinkItem | None
